tests/test_custom_metadata_fields_ops.py

```diff
@@ -0,0 +1,761 @@
+            )
+            self.client.get_custom_metadata_fields(True)
+            self.assertRaises(ForbiddenException)
+        except UnknownException as e:
+            self.assertEqual(e.message, "Your account cannot be authenticated.")
+            self.assertEqual(403, e.response_metadata.http_status_code)
+
+    @responses.activate
+    def test_get_custom_metadata_fields_succeeds(self):
+        """
+        Tests if get_custom_metadata_fields succeeds
+        """
+        URL.API_BASE_URL = "http://test.com"
+        url = "{}/v1/customMetadataFields".format(URL.API_BASE_URL)
+        headers = create_headers_for_test()
+        responses.add(
+            responses.GET,
+            url,
+            body=json.dumps([{
+                "id": "62a9d5f6db485107347bb7f2",
+                "name": "test10",
+                "label": "test10",
+                "schema": {
+                    "type": "Number",
+                    "isValueRequired": False,
+                    "minValue": 10,
+                    "maxValue": 1000
+                }
+            }, {
+                "id": "62aab2cfdb4851833b8f5e64",
+                "name": "test11",
+                "label": "test11",
+                "schema": {
+                    "type": "Number",
+                    "isValueRequired": False,
+                    "minValue": 10,
+                    "maxValue": 1000
+                }
+            }]),
+            match=[matchers.query_string_matcher("includeDeleted=False")],
+            headers=headers
+        )
+        resp = self.client.get_custom_metadata_fields()
+
+        mock_response_metadata = {
+            'raw': [{
+                'id': '62a9d5f6db485107347bb7f2',
+                'name': 'test10',
+                'label': 'test10',
+                'schema': {
+                    'type': 'Number',
+                    'isValueRequired': False,
+                    'minValue': 10,
+                    'maxValue': 1000
+                }
+            }, {
+                'id': '62aab2cfdb4851833b8f5e64',
+                'name': 'test11',
+                'label': 'test11',
+                'schema': {
+                    'type': 'Number',
+                    'isValueRequired': False,
+                    'minValue': 10,
+                    'maxValue': 1000
+                }
+            }],
+            'httpStatusCode': 200,
+            'headers': {
+                'Content-Type': 'text/plain',
+                'Accept-Encoding': 'gzip, deflate',
+                'Authorization': 'Basic ZmFrZTEyMjo='
+            }
+        }
+
+        self.assertEqual(camel_dict_to_snake_dict(mock_response_metadata), resp.response_metadata.__dict__)
+        self.assertEqual('62a9d5f6db485107347bb7f2', resp.list[0].id)
+        self.assertEqual('62aab2cfdb4851833b8f5e64', resp.list[1].id)
+        self.assertEqual("http://test.com/v1/customMetadataFields?includeDeleted=False", responses.calls[0].request.url)
```

Comment:
Regarding your 2nd point about not sending includeDelete, I think it'll not cause any bug. And it is more convenient to do this right now because we can then make use of python's support for default arguments feature as is done here:
https://github.com/amanforindia/imagekit-python/blob/0a958ee917f97d0cf7adcebb7b5ad0ab8395ab34/imagekitio/client.py#L222

Reply:
This is fine. However test case for `includeDeleted=true` is missing.